robot-server/robot_server/health/router.py

```diff
@@ -9,8 +13,78 @@
 from robot_server.service.legacy.models import V1BasicResponse
 from .models import Health, HealthLinks
 
+_log = logging.getLogger(__name__)
 
 LOG_PATHS = ["/logs/serial.log", "/logs/api.log", "/logs/server.log"]
+VERSION_PATH = "/etc/VERSION.json"
+
+
+@dataclass
+class ComponentVersions:
+    """Holds the versions of system components."""
+
+    api_version: str
+    system_version: str
+
+
+def _get_version() -> Dict[str, str]:
+    try:
+        with open(VERSION_PATH, "r") as version_file:
+            return cast(Dict[str, str], json.load(version_file))
+    except FileNotFoundError:
+        _log.warning(f"{VERSION_PATH} does not exist - is this a dev server?")
+        return {}
+    except OSError as ose:
+        _log.warning(
+            f"Could not open {VERSION_PATH}: {ose.errno}: {ose.strerror} - is this a dev server?"
+        )
+        return {}
+    except json.JSONDecodeError as jde:
+        _log.error(
+            f"Could not parse {VERSION_PATH}: {jde.msg} at line {jde.lineno} col {jde.colno}"
+        )
+        return {}
```

Comment:
If it makes it easier, instead of constructing a message out of `.errno` and such, I think you can do `exc_info=True` to get the logger to include the full details of the current exception.

```suggestion
    except OSError as ose:
        _log.warning(
            f"Could not open {VERSION_PATH} - is this a dev server?",
            exc_info=True,
        )
        return {}
    except json.JSONDecodeError as jde:
        _log.error(
            f"Could not parse {VERSION_PATH}",
            exc_info=True,
        )
        return {}
```

Reply:
> It's surprising to me that the version string is semantically meaningful enough to cause any kind of breakage if it says `a` instead of `alpha`. I'll have to catch up on the details of the bug, but it seems like we should either not have the app assume any specific format, or make it really impossible for the server to return anything outside that specific format.

Yeah this is because [node-semver](https://github.com/npm/node-semver) doesn't like that format. Maybe there's a way to get it to like that format, and probably we should also do that, but this makes the robot server more compliant with the update server anyway.

Reply:
Yeah, I don't really like that because it makes it a little clearer what you do and don't get with the explicit formatting. Plus it makes sure we don't get a bunch of big tracebacks when we already know where the problem is.